src/nlpviewer/components/TextViewer.tsx

```diff
@@ -80,139 +84,260 @@ function TextViewer({ plugins, onEvent }: TextViewerProp) {
   const selectedLink = links.find(link => link.id === selectedLinkId) || null;
   const enabledPlugins = plugins.filter(p => p.enabled(appState));
 
-  return (
-    <div className={style.text_viewer}>
-      <main className={style.layout_container}>
-        <div className={style.metadata_side_container}>
-          <TextDetail
-            annotationLegends={annotationLegendsWithColor}
-            linkLegends={linksLegendsWithColor}
-            attributes={attributes}
-            ontology={ontology}
-          />
+  const pluginsByName = new Map<string, IPlugin>(
+    enabledPlugins.map(
+      p => [p.name, p]
+    )
+  );
+
+  var pluginKey = 0
+
+  function renderPlugin(p:IPlugin){
+    const Comp = p.component;
+    pluginKey = pluginKey + 1;
+    return <Comp key={pluginKey} dispatch={dispatch} appState={appState}/>;
```

Comment:
plugin name could be a better one for `key` than a global variable `pluginKey `

Reply:
Done